library/kotlin/io/envoyproxy/envoymobile/EngineBuilder.kt

```diff
@@ -162,8 +164,8 @@ open class EngineBuilder(
    *
    * @return this builder.
    */
-  fun addStringAccessor(name: String, accessor: EnvoyStringAccessor): EngineBuilder {
-    this.stringAccessors.put(name, accessor)
+  fun addStringAccessor(name: String, accessor: () -> ByteBuffer): EngineBuilder {
```

Comment:
Why is this configured to accept a `ByteBuffer` instead of a `String` like iOS has?

Reply:
Because we have existing utility functions to bridge ByteBuffer across the JNI.

Reply:
IMO we should keep consistency between the public iOS/Android interfaces here (especially since the `name` parameter on this line also accepts a `String`). It would also help give consumers some guardrails to ensure they don't pass a dataset of something that is not a string. Thoughts?